packages/manager/cypress/e2e/core/cloudpulse/widget-level-dimension-filter.spec.ts

```diff
@@ -0,0 +1,682 @@
+        ui.button
+          .findByAttribute('label', 'Cancel')
+          .should('be.visible')
+          .and('be.enabled');
+        ui.drawerCloseButton.find().should('be.visible').and('be.enabled');
+        cy.get('[data-qa-dimension-filter="dimension_filters.0-id"]').should(
+          'not.exist'
+        );
+      });
+
+    // Define filters to add
+    const filtersWithOperators = [
+      {
+        dimension: 'entity_id',
+        operator: 'Not Equal',
+        valueToSelect: '345',
+        apiOperator: 'neq',
+        apiValue: '345',
+        valuePlaceHolder: 'Enter a Value',
+        isAutocomplete: false,
+      },
+      {
+        dimension: 'response_type',
+        operator: 'Starts with',
+        valueToSelect: '2x',
+        apiOperator: 'startswith',
+        apiValue: '2x',
+        valuePlaceHolder: 'Enter a Value',
+        isAutocomplete: false,
+      },
+      {
+        dimension: 'response_type',
+        operator: 'Ends with',
+        valueToSelect: '5xx',
+        apiOperator: 'endswith',
+        apiValue: '5xx',
+        valuePlaceHolder: 'Enter a Value',
+        isAutocomplete: false,
+      },
+      {
+        dimension: 'Protocol',
+        operator: 'In',
+        valueToSelect: 'Select All',
+        apiOperator: 'in',
+        apiValue: 'TCP,UDP',
+        valuePlaceHolder: 'Select Values',
+        isAutocomplete: true,
+      },
+      {
+        dimension: 'linode_id',
+        operator: 'Equal',
+        valueToSelect: 'Test-linode-3',
+        apiOperator: 'eq',
+        apiValue: '789',
+        valuePlaceHolder: 'Select a Value',
+        isAutocomplete: true,
+      },
+    ];
+
+    // Add filters one by one
+    filtersWithOperators.forEach(
+      (
+        {
+          dimension,
+          operator,
+          valueToSelect,
+          valuePlaceHolder,
+          isAutocomplete,
+        },
+        index
+      ) => {
+        // Click Add Filter button
+        ui.button.findByTitle('Add Filter').click();
+        const valueSelector = `[data-qa-dimension-filter="dimension_filters.${index}-value"]`;
+
+        cy.get(`[data-testid="dimension_filters.${index}-id"]`).within(() => {
+          // Select dimension
+          ui.autocomplete.findByLabel('Dimension').should('be.visible').click();
+          ui.autocomplete.findByLabel('Dimension').type(dimension);
+
+          ui.autocompletePopper
+            .findByTitle(dimension)
+            .should('be.visible')
+            .click();
+
+          // Select operator
+          ui.autocomplete
+            .findByLabel('Operator')
+            .should('be.visible')
+            .type(operator);
+
+          ui.autocompletePopper
+            .findByTitle(operator)
+            .should('be.visible')
+            .click();
+
+          cy.get(valueSelector)
+            .findByPlaceholderText(valuePlaceHolder)
+            .should('be.visible');
+          cy.get(valueSelector).findByPlaceholderText(valuePlaceHolder).click();
```

Comment:
```suggestion
          cy.get(valueSelector)
            .findByPlaceholderText(valuePlaceHolder)
            .should('be.visible').click()
```

both can go in one shot right

Reply:
we do  like this also   cy.get(valueSelector)
  .findByPlaceholderText(valuePlaceHolder)
  .as('valueInput')
  .should('be.visible');

cy.get('@valueInput').click();